LibCarla/source/carla/opendrive/OpenDrive.h

```diff
@@ -6,18 +6,22 @@
 
 #pragma once
 
+#include "carla/Memory.h"
 #include "carla/road/Map.h"
+#include "parser/OpenDriveParser.h"
 
 #include <istream>
 #include <ostream>
+#include <string>
 
 namespace carla {
 namespace opendrive {
 
     class OpenDrive {
     public:
 
-      static road::Map Load(std::istream &input);
+      static SharedPtr<road::Map> Load(std::istream &input, std::string *out_error = nullptr);
```

Comment:
Is this common in carla to use out variables like this for errors (or use exception?)

Reply:
Not really, here we could just return null and log internally. (As for exceptions we have the issue that Unreal compiles without exceptions, we'd like to stop using exceptions too in the server-side).